src/CLI/Encrypt_Decrypt/main.cpp

```diff
@@ -1,417 +1,15 @@
-#include "../../../src/lib/SoftHSM.h"
-//#include <corecrt_io.h>// Include the correct SoftHSM header
-#include <fcntl.h>
-#include <cstdlib>
-#include <iostream>
-#include <vector>
-#include <fstream>
-#include <filesystem>
-#include <cstring>
 
-typedef enum
-{
-	/** Operation was successful */
-	kStatus_SSS_Success = 0x5a5a5a5au,
-	/** Operation failed */
-	kStatus_SSS_Fail = 0x3c3c0000u,
-	/** Operation not performed because some of the passed parameters
-	 * were found inappropriate */
-	kStatus_SSS_InvalidArgument = 0x3c3c0001u,
-	// LCOV_EXCL_START
-	/** Where the underlying sub-system supports multi-threading,
-	 * Internal status to handle simultaneous access.
-	 *
-	 * This status is not expected to be returned to higher layers.
-	 * */
-	kStatus_SSS_ResourceBusy = 0x3c3c0002u,
-	// LCOV_EXCL_STOP
-} sss_status_t;
-
-namespace fs = std::filesystem;
-
-#define FILL_ATTR(attr, typ, val, len) {(attr).type=(typ); (attr).pValue=(val); (attr).ulValueLen=len;}
-#define S_IRUSR 0400
-#define S_IWUSR 0200
-
-#define MAGIC "35"
-#define GET_KEK_BT_INFO "105"
-#define ENCRYPT_DATA_KEY "107"
-#define DECRYPT_DATA_KEY "108"
-
-// Create an instance of SoftHSM
-SoftHSM* hsm = SoftHSM::i(); // Get the instance
-
-void intToBytes(int value, unsigned char* bytes) {
```

Comment:
1. I do not understand what purpose ? big little endian?
2.  anyway, why not using smart pointer and pass it by reference?